lib/runners/processor/actionlint.rb

```diff
@@ -0,0 +1,109 @@
+module Runners
+  class Processor::Actionlint < Processor
+    SCHEMA = _ = StrongJSON.new do
+      extend Schema::ConfigTypes
+
+      # @type self: SchemaClass
+      let :config, base(
+        target: target,
+
+        # Native options
+        ignore: one_or_more_strings?,
+      )
+
+      let :issue, object(
+        shellcheck: object?(id: string, severity: string, line: integer, column: integer, message: string, url: string),
+      )
+    end
+
+    ALLOWED_ERRORS = [
+      "no project was found in any parent directories of",
+      "no YAML file was found in",
+    ].freeze
+    SHELLCHECK_MESSAGE_FORMAT = /shellcheck reported issue in this script: (?<id>SC[^:]+):(?<severity>\w+):(?<line>\d+):(?<column>\d+): (?<message>.*)$/.freeze
+
+    register_config_schema SCHEMA.config
+
+    def self.config_example
+      <<~'YAML'
+        root_dir: project/
+        target:
+          - .github/workflows/one_workflow.yml
+          - .github/workflows/another_workflow.yml
+        ignore:
+          - "job .* needs job"
+          - "unknown Webhook event"
+      YAML
+    end
+
+    # This analyser uses git metadata (.git/).
+    def use_git_metadata_dir?
+      true
+    end
+
+    def analyzer_options
+      [
+        "-format", "{{json .}}",
+
+        # Specify error message patterns to ignore
+        *(Array(config_linter[:ignore]).flat_map { |v| ["-ignore", v] }),
+
+        # After -- separator, specify target files to analyze
+        "--", *Array(config_linter[:target]),
+      ]
+    end
+
+    def analyze(changes)
+      stdout, stderr, status = capture3(analyzer_bin, *analyzer_options)
+
+      # @see https://github.com/rhysd/actionlint/blob/v1.6.3/docs/usage.md#exit-status
+      case
+      when [0, 1].include?(status.exitstatus)
+        # Zero or more issue found => Success
+        Results::Success.new(guid: guid, analyzer: analyzer, issues: parse_result(stdout))
+      when ALLOWED_ERRORS.any? { |pattern| stderr.include?(pattern) }
+        # No yaml files found => Success
+        add_warning(stderr)
+        Results::Success.new(guid: guid, analyzer: analyzer)
+      else
+        # Other outcomes (e.g. fatal errors) with exit code 2, 3 => Failure
+        add_warning(stderr)
+        raise "Actionlint returned exit code #{status.exitstatus}. See logs for details."
```

Comment:
[trivial] Can we use `analyzer_name`?

```suggestion
        raise "#{analyzer_name} returned exit code #{status.exitstatus}. See logs for details."
```

Reply:
Ah I missed this. Thank you.